packages/dd-trace/src/exporters/common/request.js

```diff
@@ -93,16 +95,36 @@ function request (data, options, callback) {
   options.agent = isSecure ? httpsAgent : httpAgent
 
   const onResponse = res => {
+    const isGzip = res.headers['content-encoding'] === 'gzip'
+    const chunks = []
     let responseData = ''
 
     res.setTimeout(timeout)
 
-    res.on('data', chunk => { responseData += chunk })
+    res.on('data', chunk => {
+      if (isGzip) {
+        chunks.push(chunk)
+      } else {
+        responseData += chunk
+      }
+    })
     res.on('end', () => {
       activeRequests--
 
       if (res.statusCode >= 200 && res.statusCode <= 299) {
-        callback(null, responseData, res.statusCode)
+        if (isGzip) {
+          const buffer = Buffer.concat(chunks)
+          zlib.gunzip(buffer, (err, result) => {
+            if (err) {
+              log.error(`Could not gunzip response: ${err.message}`)
+              callback(null, '', res.statusCode)
+            } else {
+              callback(null, result.toString(), res.statusCode)
```

Comment:
Can we presume utf-8 here for character encoding?

I _guess_ in the non-gzip case the `res` object would produce a `chunk` that's already a properly decoded string based on the (optional) charset in the response's `Content-type` header. Here, though, we might need to heed the charset declaration ourselves. This FWIW is not even trivial in case we need to presume a default, as different content types have different defaults. Maybe we already have some prior art code somewhere for this?

Reply:
> Can we presume utf-8 here for character encoding?

yeah, as far as I know, the only data that we ever receive is `application/json`, which is encoded using utf-8